libs/optimizer/src/lib/utils/concatTracks.ts

```diff
@@ -0,0 +1,35 @@
+import { ScoringTrack } from '../optimizer';
+
+/**
+ * concatenate multiple tracks into one track.
+ * it removes duplicated points between each end and start of consecutive tracks
+ * @param tracks the tracks to concatenate
+ * @return a track
+ */
+export function concatTracks(...tracks: ScoringTrack[]): ScoringTrack {
+  const concatenated: ScoringTrack = {
+    lat: [],
+    lon: [],
+    alt: [],
+    timeSec: [],
+    minTimeSec: 0,
+  };
+
+  function concat(track: ScoringTrack) {
```

Comment:
suggest:

```typescript
 for (const track of tracks) {
    if (track.lat.length == 0) {
      continue;
    }
    const skipFirstPoint = concatenated.lat.at(-1) === track.lat[0]  && concatenated.lat.at(-1) === track.lat[0];
    for (const key of ['lat', 'lon', 'alt', 'timeSec']) {
      const arrayToConcat = skipFirstPoint ? track[key].slice(1): track[key]
      concatenated[key] = concatenated[key].concat(arrayToConcat);
    }
  }
```

Reply:
Much more readable indeed. I added this suggestion